src/bitcoin-cli.cpp

```diff
@@ -97,11 +97,19 @@ static bool AppInitRPC(int argc, char* argv[])
 
 UniValue CallRPC(const string& strMethod, const UniValue& params)
 {
-    if (mapArgs["-rpcuser"] == "" && mapArgs["-rpcpassword"] == "")
-        throw runtime_error(strprintf(
-            _("You must set rpcpassword=<password> in the configuration file:\n%s\n"
-              "If the file does not exist, create it with owner-readable-only file permissions."),
-                GetConfigFile().string().c_str()));
+    std::string strRPCUserColonPass;
+    if (mapArgs["-rpcpassword"] == "") {
+        /* Try fall back to cookie-based authentication if no password is provided */
+        if (!GetAuthCookie(&strRPCUserColonPass)) {
+            throw runtime_error(strprintf(
```

Comment:
I think this if block from L100-L112 should go down to L126 (after the connection has been set up).
It would make more sense to first check if we can connect to bitcoind, then check if we have some credentials available.